Sources/Identity/CustomerInfo.swift

```diff
@@ -360,15 +360,14 @@ private extension CustomerInfo {
         return Set(
             dates
                 .lazy
-                .filter {
-                    guard let date = $1 else { return true }
-                    return self.isAfterReferenceDate(date: date)
-                }
+                .filter { self.isDateActive($1) }
                 .map { key, _ in key }
         )
     }
 
-    func isAfterReferenceDate(date: Date) -> Bool { date.timeIntervalSince(self.requestDate) > 0 }
+    private func isDateActive(_ date: Date?) -> Bool {
+        return EntitlementInfo.isDateActive(expirationDate: date, for: self.requestDate)
```

Comment:
This is a bit of a nitpick, but we are using a method in `EntitlementInfo` for subscriptions, which is not related to entitlements at all.

Reply:
Yeah that's fair, let me move this somewhere else.